From 0cc8b2cfd3900621a5d32719be29f0d1cc65e5c0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Alex=20R=C3=B8nne=20Petersen?= <alex@alexrp.com>
Date: Fri, 17 Oct 2025 14:58:30 +0200
Subject: [PATCH 011/133] libunwind: backport llvm/llvm-project#162867
X-Developer-Signature: v=1; a=openpgp-sha256; l=1003; i=xtex@astrafall.org;
 h=from:subject; bh=F4ndrq+NknZZ8V/CWlWVcac7APNkwWnOzGVlxvD7Jhg=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn7cn+mTS2a1JOW9XZ5NqdvYJ7Ituym9L8iyXLdb1UE5
 x2pLWDqmMjCIMbFYCmmyFJk2ODNqpPOL7qsXBZmDisTyBBpkQYGBgYGFga+3MS8UiMdIz1TbUM9
 Q0MdIx0jBi5OAZjqwhSGv0IiXxs0mtuWMNSkCua8M7JwMPmwV+DhjQSrR3n3m5qNGBkZ2t8ufig
 WqPG/VSzN2pJbMCf8w7EpTaeS3HTcLl/u3RHODAA=
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

https://github.com/llvm/llvm-project/pull/162867
(cherry picked from commit 671428359ef85d5ffc9e44fbc8bd51ac0079f917)
---
 lib/libunwind/src/Unwind-seh.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/libunwind/src/Unwind-seh.cpp b/lib/libunwind/src/Unwind-seh.cpp
index b2bb119ed6d2..ad3bc3048b80 100644
--- a/lib/libunwind/src/Unwind-seh.cpp
+++ b/lib/libunwind/src/Unwind-seh.cpp
@@ -148,7 +148,8 @@ _GCC_specific_handler(PEXCEPTION_RECORD ms_exc, PVOID frame, PCONTEXT ms_ctx,
     }
     // FIXME: Indicate target frame in foreign case!
     // phase 2: the clean up phase
-    RtlUnwindEx(frame, (PVOID)disp->ControlPc, ms_exc, exc, ms_ctx, disp->HistoryTable);
+    RtlUnwindEx(frame, (PVOID)disp->ControlPc, ms_exc, exc, disp->ContextRecord,
+                disp->HistoryTable);
     _LIBUNWIND_ABORT("RtlUnwindEx() failed");
   case _URC_INSTALL_CONTEXT: {
     // If we were called by __libunwind_seh_personality(), indicate that
-- 
2.52.0

